### ISSUE TITLE
Case changes on a case insensitive file system

### ISSUE THREAD

**Issue by mried** [sampsyo/beets#1496]
I've got the following album: "Sounds Of The Universe". While importing it with `move` set to `true` it should be renamed to "Sounds of the Universe" (so only the case has changed). This will work on a case sensitive file system since a second folder is created first and the contents is then copied into it. On a case insensitive file system, no new folder will be created hence the files won't be moved. But at the database, the new path is stored.
This is not what I expect beets to do. I tried to fix this but I'm completely unsure how to do this in a elegant way because it should work on a case sensitive file system, too. I guess the right place is `Item.move()` defined in `library.py`.
Any suggestions?

**Comment by sampsyo** [comment]
Interesting point. What would the correct behavior be, even if we ignored case-sensitive filesystems? It seems really hard to get right—we can't "copy" files from path A to path B in this situation, since we would need to get rid of the conflicting directory in path A before files can safely arrive at path B (and we don't know that path A's directory will be emptied out, so it's not safe to remove it).

Perhaps the path would just not change?

**Comment by mried** [comment]
Exactly these points stopped me from fixing it :wink:
The only solution I can think of is to *rename* the folder / file. `os.rename` is able to rename a folder `a` to `A` but this has to be done recursively. This sounds possible. But what happens on a case sensitive file system? What if there is already a folder named `A`?
The worst thing about it is the path stored at the database. To use the example I provided above: I've got the path which uses capital letters: "Sounds Of The Universe". Now I search for it like so:
```
beet ls path:"/music/Sounds Of The Universe""
```
This will return nothing since the path at the library is "Sounds of the Universe". Thats really confusing (that's why I opened #1491...)
